app/models/concerns/course/assessment/submission/workflow_event_concern.rb

```diff
@@ -68,11 +68,17 @@ def submission_graded_email_enabled?
   # Defined outside of the workflow transition as points_awarded and draft_points_awarded are
   # not set during the event transition, hence they are not modifiable within the method itself.
   def assign_experience_points
-    # publish event (from grade) - Deduce points awarded from draft or updated attribute.
-    if workflow_state == 'published' &&
-       (workflow_state_was == 'graded' || workflow_state_was == 'submitted')
-      self.points_awarded ||= draft_points_awarded
-      self.draft_points_awarded = nil
+    if workflow_state == 'published'
+      # publish event (from grade) - Deduce points awarded from draft or updated attribute.
+      if workflow_state_was == 'graded' || workflow_state_was == 'submitted'
+        self.points_awarded ||= draft_points_awarded
+        self.draft_points_awarded = nil
+      # finalise event (from attempting) - Assign 0 points as there are no questions.
+      elsif workflow_state_was == 'attempting'
+        self.points_awarded = 0
```

Comment:
Although we are somewhat assigning experience points, I'd prefer this logic in the `finalise` method instead, it would be much easier to read through and isolate the logic to set the necessary fields (points_awarded, awarded_at and awarder). 

The reason for having the original method was because it was abstracted from 2 workflow events.